composer/datasets/synthetic.py

```diff
@@ -125,3 +128,104 @@ def __getitem__(self, idx: int):
             return self.transform(self.input_data[idx]), self.input_target[idx]
         else:
             return self.input_data[idx], self.input_target[idx]
+
+
+class SyntheticPILDataset(VisionDataset):
```

Comment:
Thanks for writing this! This pretty implements #346 :)

Jotting this down for another issue/PR, unless if you have the bandwidth to implement it tmrw :):

A lot of this logic seems very similar to the `SyntheticBatchPairDataset`.

Could the `SyntheticPILDataset` internally use a `SyntheticBatchPairDataset` to generate the tensors, and then wrap it in a PIL image (and apply the transformations) on `__getitem__`.? I think that would clean up the code.

Reply:
Oh wow, nice to have inadvertently resolved an outstanding issue :)
I'm not quite sure how to internally use a `SyntheticBatchPairDataset` to generate the data for `SyntheticPILDataset` because `SyntheticBatchPairDataset` generates the tensors upon the first call to `__getitem__`, and `SyntheticPILDataset` would need to override `__getitem__`. It might be possible to change `SyntheticBatchPairDataset` to generate the data on init, but the comment for `__getitem__` in `SyntheticBatchPairDataset` makes it seem as though the data need to be generated upon the first call to `__getitem__` so that they're on the correct gpu.